src/EnergyPlus/Autosizing/BaseSizerWithScalableInputs.cc

```diff
@@ -155,6 +156,65 @@ void BaseSizerWithScalableInputs::initializeWithinEP(EnergyPlusData &state,
+    Real64 constexpr RatedAmbAirTemp = 35.0;            // 95 F cooling mode
+    Real64 constexpr RatedInletWaterTempHeat = 21.1111; // 21.11C or 70F, heating mode
+    Real64 constexpr RatedAmbAirTempHeat = 8.3333;      // 8.33 or 47F, heating mode
+
+    // this is not correct at the moment. Added in #9457 and will be corrected in #9439 or other PR
+    switch (coilType) {
+    case DataHVACGlobals::Coil_CoolingWaterToAirHPVSEquationFit: {
+        sourceTemp = RatedInletWaterTemp;
+        // should be plant supply node temperature
+    } break;
+    case DataHVACGlobals::Coil_HeatingWaterToAirHPVSEquationFit: {
+        sourceTemp = RatedInletWaterTempHeat;
+        // should be plant supply node temperature
+    } break;
+    case DataHVACGlobals::CoilDX_HeatPumpWaterHeaterVariableSpeed: {
+        // RatedSourceTemp = state.dataVariableSpeedCoils->VarSpeedCoil(CoilIndex).WHRatedInletWaterTemp;
+        sourceTemp = RatedInletWaterTempHeat;
+        // should be HPWH tank node set point temperature
+    } break;
+    case DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed: {
+        // should use outdoor temp at zone peak
+        // if (this->curZoneEqNum > 0) {
+        //    // should be outdoor temp at zone peak
+        //    int TimeStepNumAtMax = this->finalZoneSizing(this->curZoneEqNum).TimeStepNumAtCoolMax;
+        //    int DDNum = this->finalZoneSizing(this->curZoneEqNum).CoolDDNum;
+        //    if (DDNum > 0 && TimeStepNumAtMax > 0) {
+        //        sourceTemp = state.dataSize->DesDayWeath(DDNum).Temp(TimeStepNumAtMax);
+        //    }
+        //} else if (this->curSysNum > 0) {
+        //    // should be outdoor temp at system peak
+        //    sourceTemp = state.dataSize->FinalSysSizing(this->curSysNum).OutTempAtCoolPeak;
+        //}
+        sourceTemp = RatedAmbAirTemp; // overwrite for time being until this gets fixed
+    } break;
+    case DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed: {
+        // should use outdoor temp at zone peak
+        // if (this->curZoneEqNum > 0) {
+        //    int TimeStepNumAtMax = this->finalZoneSizing(this->curZoneEqNum).TimeStepNumAtHeatMax;
+        //    int DDNum = this->finalZoneSizing(this->curZoneEqNum).HeatDDNum;
+        //    if (DDNum > 0 && TimeStepNumAtMax > 0) {
+        //        sourceTemp = state.dataSize->DesDayWeath(DDNum).Temp(TimeStepNumAtMax);
+        //    }
+        //} else if (this->curSysNum > 0) {
+        //    // should be outdoor temp at system peak
+        //    sourceTemp = state.dataSize->FinalSysSizing(this->curSysNum).HeatOutTemp;
+        //}
+        sourceTemp = RatedAmbAirTempHeat; // overwrite for time being until this gets fixed
+    } break;
+    default: {
+        // catches all parent objects that do not set coilType, do nothing
+        // catches all unit tests that set a different source temp without setting coilType, do nothing
+    } break;
+    }
+    return sourceTemp;
+}
+
```

Comment:
This new sizing member function mimics the VS coil model function used to set the coil source temperature (e.g., OAT or entering water T). That function will need to be modified in the near future. This function sets up a method for a parent object (i.e., UnitarySystem) to gather the same information used for sizing calculations. There will obviously be more changes here as the VS coil model sizing calculations evolve.